http-client/src/test/java/io/airlift/http/client/jetty/TestJettyAsyncExecutorHttpClient.java

```diff
@@ -0,0 +1,55 @@
+package io.airlift.http.client.jetty;
+
+import io.airlift.http.client.AbstractHttpClientTest;
+import io.airlift.http.client.HttpClientConfig;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.ResponseHandler;
+import io.airlift.http.client.StreamingResponse;
+
+import java.util.Optional;
+import java.util.concurrent.Executors;
+
+import static io.airlift.concurrent.Threads.virtualThreadsNamed;
+
+public class TestJettyAsyncExecutorHttpClient
+        extends AbstractHttpClientTest
+{
+    @Override
+    protected HttpClientConfig createClientConfig()
+    {
+        return new HttpClientConfig()
+                .setHttp2Enabled(false);
+    }
+
+    @Override
+    public Optional<StreamingResponse> executeRequest(AbstractHttpClientTest.CloseableTestHttpServer server, Request request)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void testConnectTimeout()
+            throws Exception
+    {
+        // Async client doesn't work properly with timeouts as this is expected for the caller to put timeout on the Future.get
+        doTestConnectTimeout(true);
```

Comment:
That comment is worrisome. There should be two timeouts: on the request itself and the caller's on future.get

also, do we test that caller's `future.cancel` gets propagated?